### ISSUE TITLE
VirtualBox Fails to Buils

### ISSUE THREAD

**Issue by wjjunyor** [NixOS/nixpkgs#101823]
**While building VirtualBox with Extension Pack **
I am trying to add VirtualBox via configuration.nix, but while the switch, the nixos-build returns the following error: 
```In file included from /build/virtualbox-6.1.14-modsrc/vboxdrv/linux/SUPDrv-linux.c:33:
/build/virtualbox-6.1.14-modsrc/vboxdrv/r0drv/linux/the-linux-kernel.h:141:11: fatal error: linux/smp_lock.h: No such file or directory
  141 | # include <linux/smp_lock.h>
      |           ^~~~~~~~~~~~~~~~~~
```

**To Reproduce**
1. On NixOs 21.03pre248308.5d0e2d55
2. add the following to your configuration.nix: 
```
  # Virtualization activation fos this user
   virtualisation.virtualbox.host.enable = true;
   virtualisation.virtualbox.host.enableExtensionPack = true;
   users.extraGroups.vboxusers.members = [ "youruser" ];
``` 
3. Run: `sudo nixos-rebuild switch --upgrade`

**Expected behavior**
Instead of a complete successful rebuild, the nixos-rebuild command returns: `error: build of '/nix/store/xaalq56p55l35dacjbwhqigvlf50c3q7-nixos-system-T101-21.03pre248308.5d0e2d55.drv' failed`

**Notify maintainers**
Please @svanderburg are you aware? Is this an issue?

**Metadata**
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result. **No results**.

**Comment by bryanasdev000** [comment]
> **While building VirtualBox with Extension Pack **
> I am trying to add VirtualBox via configuration.nix, but while the switch, the nixos-build returns the following error:
> 
> ```
> /build/virtualbox-6.1.14-modsrc/vboxdrv/r0drv/linux/the-linux-kernel.h:141:11: fatal error: linux/smp_lock.h: No such file or directory
>   141 | # include <linux/smp_lock.h>
>       |           ^~~~~~~~~~~~~~~~~~
> ```
> 
> **To Reproduce**
> Steps to reproduce the behavior:
> 
> 1. On NixOs 21.03pre248308.5d0e2d55
> 2. add the following to your configuration.nix:
> 
> ```
>   # Virtualization activation fos this user
>    virtualisation.virtualbox.host.enable = true;
>    virtualisation.virtualbox.host.enableExtensionPack = true;
>    users.extraGroups.vboxusers.members = [ "youruser" ];
> ```
> 
> 1. Run: `sudo nixos-rebuild switch --upgrade`
> 
> **Expected behavior**
> Instead of a complete successful rebuild, the nixos-rebuild command returns: `error: build of '/nix/store/xaalq56p55l35dacjbwhqigvlf50c3q7-nixos-system-T101-21.03pre248308.5d0e2d55.drv' failed`
> 
> **Notify maintainers**
> Please @svanderburg are you aware? Is this an issue?
> 
> **Metadata**
> Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result. **No results**.

It is already a known problem and that was fixed in version 6.1.16 of Virtualbox, remembering that in theory it only affects the Kernel 5.9.X. We need a PR to update VirtualBox, I'll see if I do one.

**Comment by wjjunyor** [comment]
Thanks a lot!
